freqtrade/freqtradebot.py

```diff
@@ -225,25 +225,23 @@ def process(self) -> None:
         strategy_safe_wrapper(self.strategy.bot_loop_start, supress_error=True)(
             current_time=datetime.now(timezone.utc))
 
-        self.strategy.analyze(self.active_pair_whitelist)
+        open_trade_pairs_analyzed = []
+        if self.strategy.position_adjustment_enable:
+            open_trade_pairs = [ot.pair for ot in trades]
+            self.strategy.analyze(open_trade_pairs)
 
-        with self._exit_lock:
-            # Check for exchange cancelations, timeouts and user requested replace
-            self.manage_open_orders()
+            open_trade_pairs_analyzed = open_trade_pairs
+            self.manage_open_trades()
 
-        # Protect from collisions with force_exit.
-        # Without this, freqtrade may try to recreate stoploss_on_exchange orders
-        # while exiting is in process, since telegram messages arrive in an different thread.
-        with self._exit_lock:
-            trades = Trade.get_open_trades()
-            # First process current opened trades (positions)
-            self.exit_positions(trades)
-
-        # Check if we need to adjust our current positions before attempting to buy new trades.
-        if self.strategy.position_adjustment_enable:
+            # Check if we need to adjust our current positions before attempting to buy new trades.
             with self._exit_lock:
                 self.process_open_trade_positions()
 
+        remaining_pairs_to_analyse = [pair for pair in self.active_pair_whitelist if pair not in open_trade_pairs_analyzed]
+
+        self.strategy.analyze(remaining_pairs_to_analyse)
+        self.manage_open_trades()
```

Comment:
Seems like you're calling `manage_open_trades` twice per iteration - which we clearly wouldn't want.

Reply:
Yes it is the case, do you think we could manage_open_trade, except the ones that was already managed. As the goal of this feature is to manage in priority open trades, it think two call is required. If you think it is not worth it I can scrap this PR and concentrate on other way to speed up live trades management.

Reply:
I don't understand the need for this call.

there is no need for this in my eyes, it'll handle the SAME trades again (there can't have been new trades opened between the first and 2nd call (except through force-entry - where it won't matter if they're handled the next iteration).